OpenRA.Mods.Common/EditorBrushes/EditorBlit.cs

```diff
@@ -178,12 +207,104 @@ void Blit(EditorBlitSource source, bool isRevert)
 			}
 		}
 
-		public void Commit() => Blit(blitSource, false);
-		public void Revert() => Blit(revertBlitSource, true);
+		public static IEnumerable<IRenderable> PreviewBlitSource(
+			EditorBlitSource blitSource,
+			MapBlitFilters filters,
+			CVec offset,
+			WorldRenderer wr)
+		{
+			var world = wr.World;
+			var map = world.Map;
+
+			var terrainRenderer = world.WorldActor.Trait<ITiledTerrainRenderer>();
+			var resourceRenderers = world.WorldActor.TraitsImplementing<IResourceRenderer>().ToArray();
+
+			var wOffset = map.CenterOfCell(CPos.Zero + offset) - map.CenterOfCell(CPos.Zero);
+
+			if (filters.HasFlag(MapBlitFilters.Terrain))
+			{
+				foreach (var (cpos, tile) in blitSource.Tiles)
+				{
+					var preview =
+						terrainRenderer.RenderPreview(
+							wr,
+							tile.TerrainTile,
+							map.CenterOfCell(cpos + offset));
+					foreach (var renderable in preview)
+						yield return renderable;
+				}
+			}
+
+			if (filters.HasFlag(MapBlitFilters.Resources))
+			{
+				foreach (var (cpos, tile) in blitSource.Tiles)
+				{
+					if (tile.ResourceLayerContents == null || tile.ResourceLayerContents.Value.Type == null)
+						continue;
+
+					var preview = resourceRenderers
+						.SelectMany(r => r.RenderPreview(
+							wr,
+							tile.ResourceLayerContents.Value.Type,
+							map.CenterOfCell(cpos + offset)));
+					foreach (var renderable in preview)
+						yield return renderable;
+				}
+			}
+
+			if (filters.HasFlag(MapBlitFilters.Actors))
+			{
+				foreach (var (_, editorActorPreview) in blitSource.Actors)
+				{
+					var preview = editorActorPreview.RenderWithOffset(wOffset);
+					foreach (var renderable in preview)
```

Comment:
```suggestion
					foreach (var renderable in preview.OrderBy(WorldRenderer.RenderableZPositionComparisonKey))
```
to fix the bib rendering. The paste buffer is rendered via `RenderAboveShroud`, which doesn't account for renderable z-offsets.

Reply:
Thanks! That does the trick!